packages/react-components/src/components/VideoGallery/useVideoStreamLifecycleMaintainer.ts

```diff
@@ -6,22 +6,45 @@ import { VideoStreamOptions, CreateVideoStreamViewResult, ScalingMode } from '..
 import { Cancellable, useCancellableTask } from '../utils/useCancellableTask';
 
 /**
- * Helper hook to maintain the video stream lifecycle. This calls onCreateStreamView and onDisposeStreamView
- * appropriately based on react lifecycle events and prop changes.
- * This also handles calls to view.update* appropriately such as view.updateScalingMode().
- *
+ * Props needed for {@link useVideoStreamLifecycleMaintainer}
  * @private
  */
-export const useVideoStreamLifecycleMaintainer = (props: {
+export interface VideoStreamLifecycleMaintainerProps {
   isStreamAvailable?: boolean;
   renderElementExists?: boolean;
   isMirrored?: boolean;
   scalingMode?: ScalingMode;
-  onCreateStreamView?: (options?: VideoStreamOptions) => Promise<void | CreateVideoStreamViewResult>;
-  onDisposeStreamView?: () => void;
-}): void => {
-  const { onCreateStreamView, onDisposeStreamView, isStreamAvailable, renderElementExists, isMirrored, scalingMode } =
-    props;
+  remoteParticipantId?: string;
+  onCreateLocalStreamView?: (options?: VideoStreamOptions) => Promise<void | CreateVideoStreamViewResult>;
+  onDisposeLocalStreamView?: () => void;
+  onCreateRemoteStreamView?: (
+    userId: string,
+    options?: VideoStreamOptions
+  ) => Promise<void | CreateVideoStreamViewResult>;
+  onDisposeRemoteStreamView?: (userId: string, options?: VideoStreamOptions) => Promise<void>;
+  isScreenSharingOn?: boolean;
+}
+
+/**
+ * Helper hook to maintain the video stream lifecycle. This calls onCreateStreamView and onDisposeStreamView
+ * appropriately based on react lifecycle events and prop changes.
+ * This also handles calls to view.update* appropriately such as view.updateScalingMode().
+ *
+ * @private
+ */
+export const useVideoStreamLifecycleMaintainer = (props: VideoStreamLifecycleMaintainerProps): void => {
```

Comment:
I'd recommend avoiding polymorphic behavior based on args.

So here, I'd recommend:
* Unexport `useVideoStreamLifecycleMaintainer`. We'll use it for the impls below.
* Add `useLocalVideoStreamLifecycleMaintainer`. This will call `useVideoStreamLifecycleMaintainer` with `onCreateStreamView: onCreateLocalStreamView` and `onDisposeStreamView: onDisposeLocalStreamView`.
* Add `useRemoteVideoStreamLifecyleMaintainer. This will also call `useVideoStreamLifecylceMaintainer`, but will need to use `useCallback` to bind the `remoteParticipantId` to the create/dispose view functions.

Reply:
Nice idea, refactored into useLocalVideoStreamLifecycleMaintainer and useRemoteVideoStreamLifecyleMaintainer